engine/src/main/java/org/terasology/rendering/dag/AmbientOcclusionPassesNode.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.terasology.config.Config;
+import org.terasology.config.RenderingConfig;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FrameBuffersManager;
+import org.terasology.rendering.world.WorldRenderer;
+
+import static org.lwjgl.opengl.GL11.GL_COLOR_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.glClear;
+import static org.terasology.rendering.opengl.OpenGLUtils.bindDisplay;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+import static org.terasology.rendering.opengl.OpenGLUtils.setViewportToSizeOf;
+
+/**
+ * TODO: Add diagram of this node
+ */
+public class AmbientOcclusionPassesNode implements Node {
+
+    @In
+    private FrameBuffersManager frameBuffersManager;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private Config config;
+
+    private RenderingConfig renderingConfig;
+    private FBO sceneOpaque;
+    private FBO ssaoBlurredFBO;
+    private FBO ssaoFBO;
+    private Material ssaoShader;
+    private Material ssaoBlurredShader;
+
+    @Override
+    public void initialise() {
+        renderingConfig = config.getRendering();
+        ssaoShader = worldRenderer.getMaterial("engine:prog.ssao");
+        ssaoBlurredShader = worldRenderer.getMaterial("engine:prog.ssaoBlur");
+    }
+
+    /**
+     * If Ambient Occlusion is enabled in the render settings, this method generates and
+     * stores the necessary images into their own FBOs. The stored images are eventually
+     * combined with others.
+     * <p>
+     * For further information on Ambient Occlusion see: http://en.wikipedia.org/wiki/Ambient_occlusion
+     */
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("rendering/ambientocclusionpasses");
```

Comment:
I'd say that PerformanceMonitor statements should be within the IF block, so that if something is disabled it doesn't get monitored.